src/main/java/com/direwolf20/buildinggadgets/common/util/tools/SetBackedPlacementSequence.java

```diff
@@ -0,0 +1,91 @@
+package com.direwolf20.buildinggadgets.common.util.tools;
+
+import com.direwolf20.buildinggadgets.api.building.IPositionPlacementSequence;
+import com.direwolf20.buildinggadgets.api.building.Region;
+import com.direwolf20.buildinggadgets.common.util.helpers.NBTHelper;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nonnull;
+import java.io.*;
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class SetBackedPlacementSequence implements IPositionPlacementSequence, Serializable {
+
+    private static final long serialVersionUID = 5752016672792062860L;
+
+    private Region boundingBox;
+    private HashSet<BlockPos> internalSet;
+
+    public SetBackedPlacementSequence(HashSet<BlockPos> internalSet, Region boundingBox) {
+        this.internalSet = internalSet;
+        this.boundingBox = boundingBox;
```

Comment:
The null check makes sense, but why should I copy the parameters? This class is intended as a functional wrapper so that a set can be used as an `IPlacementSequence`.

Reply:
If you don't copy the set then whoever holds a reference to it may mutate it and result in unwanted mutation of this PlacementSequence. If you allow these mutations to happen, there should be some notice of that.

Reply:
I wrapped it with `Collections.unmodifiableSet`, which has a similar effect and less overhead.

Reply:
This does not resolve the issue! It is not possible in Java to take full ownership of a reference, for that you'd need C++.
You are wrapping the set in Collections.unmodifiableSet, which returns an **unmodifiable view** - the backing Set might still be changed. See the documentation of which [specifically states, that only a view is returned](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collections.html#unmodifiableSet(java.util.Set))!